src/validation/statement.rs

```diff
@@ -540,6 +568,31 @@ fn validate_assignment(
     }
 }
 
+fn validate_variable_length_array(
```

Comment:
this validates variable-length-array assignemnts right? can we change the name to that?